core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -192,15 +193,24 @@ public final Object visitClearCommand(InvocationContext ctx, ClearCommand comman
                return;
             }
          }
-         // If this node has topology higher than some of the nodes and the nodes could not respond
-         // with the remote entry, these nodes are blocking the response and therefore we can get only timeouts.
-         // Therefore, if we got here it means that we have lower topology than some other nodes and we can wait
-         // for it in StateTransferInterceptor and retry the read later.
-         // TODO: These situations won't happen as soon as we'll implement 4-phase topology change in ISPN-5021
-         throw new OutdatedTopologyException("Did not get any successful response, got " + responses);
+         throw handleMissingSuccessfulResponse(responses);
       });
    }
 
+   protected static CacheException handleMissingSuccessfulResponse(Map<Address, Response> responses) {
+      if (responses.values().stream().anyMatch(UnsureResponse.class::isInstance)) {
+         // We got only unsure responses, as all nodes that were read-owners at the time when we've sent
+         // the request have progressed to newer topology. However we are guaranteed to have progressed
+         // to a topology at most one older, and can immediately retry.
+         return OutdatedTopologyException.getCachedInstance();
+      } else {
+         // Another instance when we don't get any successful response is when all owners are lost. We'll handle
+         // this later in StateTransferInterceptor, as we have to signal this to PartitionHandlingInterceptor
+         // if that's present.
+         return AllOwnersLostException.INSTANCE;
```

Comment:
If there was an exception, it would be thrown somewhere from JGroupsTransport, this code won't handle any ExceptionResponse instances.

I don't see why we need another delegation method (I would rather expose `OTE.INSTANCE` directly), but I'll comply in this PR and correct it later.

WRT PartitionHandlingInterceptor - checking "Why the hell did I get this exception" sounds prone to races to me; I prefer to signal the condition directly. We could give `AOLE` and `OTE` a common ancestor, if we want to spare some characters in `STI`.